crates/apollo-parser/src/lexer/mod.rs

```diff
@@ -379,19 +351,17 @@ impl<'a> Cursor<'a> {
                 },
                 State::BlockStringLiteralBackslash => match c {
                     '"' => {
-                        while self.eatc('"') {}
+                        // If this is \""", we need to eat 3 in total, and then continue parsing.
+                        // The lexer does not un-escape escape sequences so it's OK
+                        // if we take this path for \"", even if that is technically not an escape
+                        // sequence.
+                        if self.eatc('"') {
+                            self.eatc('"');
+                        }
```

Comment:
There's a test below to make sure `"""\""""""` is correctly lexed as a block string containing only `\"""`.